packages/errors/src/error.ts

```diff
@@ -92,21 +92,23 @@ export class SolanaError<TErrorCode extends SolanaErrorCode = SolanaErrorCode> e
         let context: SolanaErrorContext[TErrorCode] | undefined;
         let errorOptions: ErrorOptions | undefined;
         if (contextAndErrorOptions) {
-            // If the `ErrorOptions` type ever changes, update this code.
-            const { cause, ...contextRest } = contextAndErrorOptions;
-            if (cause) {
-                errorOptions = { cause };
-            }
-            if (Object.keys(contextRest).length > 0) {
-                context = contextRest as SolanaErrorContext[TErrorCode];
-            }
+            Object.entries(Object.getOwnPropertyDescriptors(contextAndErrorOptions)).forEach(([name, descriptor]) => {
+                // If the `ErrorOptions` type ever changes, update this code.
+                if (name === 'cause') {
+                    errorOptions = { cause: descriptor.value };
+                } else {
+                    if (context === undefined) {
+                        context = {} as SolanaErrorContext[TErrorCode];
+                    }
+                    Object.defineProperty(context, name, descriptor);
+                }
+            });
         }
-        const message = getErrorMessage(code, context);
+        // The `getErrorMessage` function should only get enumerable properties from the context.
+        const message = getErrorMessage(code, context ? { ...context } : undefined);
```

Comment:
This is a good extra safety, but `encodeContextObject()` already does `Object.entries()` internally, to which non-enumerable properties will be invisible. I think you should avoid the extra clone here and instead write a Jest test in `errors/src/context-test.ts` that `encodeContextObject()` doesn't print non-enumerable properties.